### PR TITLE
Allow to drop privileges at startup

### LINKED ISSUE #36293: Segmentation fault in su-exec on Linux ARM64 Docker image
> You have to provide the following information whenever possible.

```
$ docker run --rm -it clickhouse/clickhouse-server:22.3.3.44 bash
/entrypoint.sh: line 39:    10 Segmentation fault      $gosu test -f "$CLICKHOUSE_CONFIG" -a -r "$CLICKHOUSE_CONFIG"
Configuration file '/etc/clickhouse-server/config.xml' isn't readable by user with id '101'
```

> A clear and concise description of what works not as it is supposed to.

I use the latest version of the Docker image - 22.3.3.44.

**Enable crash reporting**

> If possible, change "enabled" to true in "send_crash_reports" section in `config.xml`:

config.xml is not related. 
The problem is that the Docker image has corrupted /bin/su-exec which prevents starting the server.

**How to reproduce**

On Linux ARM64 machine execute: 

1) `docker run --rm -it -v --entrypoint bash  clickhouse/clickhouse-server:22.3.3.44`
2) inside docker try to use `su-exec`
```
# su-exec test
Segmentation fault
```

**Expected behavior**

> A clear and concise description of what you expected to happen.

su-exec should print its usage text:
```
Usage: su-exec user-spec command [args]
```
instead of failing with segmentation fault

Initially reported at https://github.com/housepower/spark-clickhouse-connector/pull/85

### PR DESCRIPTION
**Pull request by alexey-milovidov**
### Changelog category (leave one):
- Build/Testing/Packaging Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Allow to drop privileges at startup. This simplifies Docker images. Closes #36293.

### CODE REVIEW

File: docker/server/Dockerfile.alpine [hunk 1]
```diff
@@ -63,7 +63,7 @@ RUN arch=${TARGETARCH:-amd64} \
     && chown clickhouse:clickhouse /var/lib/clickhouse \
     && chown root:clickhouse /var/log/clickhouse-server \
     && chmod +x /entrypoint.sh \
-    && apk add --no-cache su-exec bash tzdata \
+    && apk add --no-cache bash tzdata \
     && cp /usr/share/zoneinfo/UTC /etc/localtime \
```
Comment (alexey-milovidov) [inline docker/server/Dockerfile.alpine:67]: This looks outdated.

File: docker/server/Dockerfile.ubuntu [hunk 2]
```diff
@@ -18,13 +16,7 @@ RUN sed -i "s|http://archive.ubuntu.com|${apt_archive}|g" /etc/apt/sources.list
...
         locales \
         wget \
-        tzdata \
-    && apt-get install -y --no-install-recommends tcc libc-dev && \
-        tcc /su-exec.c -o /bin/su-exec && \
-        chown root:root /bin/su-exec && \
-        chmod 0755 /bin/su-exec && \
-        rm /su-exec.c && \
-        apt-get purge -y --auto-remove tcc libc-dev libc-dev-bin libc6-dev linux-libc-dev \
+        tzdata
```
Comment (alexey-milovidov) [inline docker/server/Dockerfile.ubuntu:19]: Looks old, we should work without `tzdata` and `locales`.

File: docker/server/Dockerfile.ubuntu [hunk 3]
```diff
@@ -3,8 +3,6 @@ FROM ubuntu:20.04
 # see https://github.com/moby/moby/issues/4032#issuecomment-192327844
 ARG DEBIAN_FRONTEND=noninteractive
 
-COPY su-exec.c /su-exec.c
-
 # ARG for quick switch to a given ubuntu mirror
 ARG apt_archive="http://archive.ubuntu.com"
```
Comment (alexey-milovidov) [inline docker/server/Dockerfile.ubuntu:7]: `archive.ubuntu.com` looks completely misleading, there is no chance find appropriate ClickHouse version there.

**Comment by alexey-milovidov** [comment]
Merging with master after #36337.

**Comment by alexey-milovidov** [comment]
Waiting for #36361

**Comment by nikitamikhaylov** [comment]
@alexey-milovidov Keeper images also can be simplified

**Comment by alexey-milovidov** [comment]
@nikitamikhaylov no review?

**Comment by alexey-milovidov** [comment]
I will unassign as one month has passed.

**Comment by alexey-milovidov** [comment]
At least now it works as expected: https://pastila.nl/?001c5a5c/93287c2e9c4f57eafe6a98962306bb52
Let's check what is wrong with the stress tests...

**Comment by alexey-milovidov** [comment]
Does not work due to

> 2. If the effective user ID is changed from 0 to nonzero, then
          all capabilities are cleared from the effective set.

**Comment by alexey-milovidov** [comment]
Amazing...

**Comment by excitoon** [comment]
Cool! It would be also awesome to have `ping` and `netstat` for more convenient network debugging inside docker images.

**Comment by alexey-milovidov** [comment]
@excitoon This `su` tool is added to solve a very specific problem with Docker and Linux.
See the comments in the code.

**Comment by pkit** [comment]
@alexey-milovidov this PR totally breaks the NFS mounts for metadata, because of root squash.
Mainly because it assumes that root access is uid=0 (while it can be uid=nobody for NFS)
Will try to roll back some of the changes

**Comment by pkit** [comment]
See https://github.com/ClickHouse/ClickHouse/pull/39898

**Pull request merged**